sound/soc/sof/intel/hda-dai.c

```diff
@@ -152,76 +152,66 @@ static int hda_link_dma_params(struct hdac_ext_stream *stream,
 }
 
 /* find the snd_sof_dai matching the BE CPU DAI name and update its config */
-static struct snd_sof_dai *hda_get_sof_dai(struct sof_intel_hda_stream *hda_stream,
-					   const char *dai_name, int channel, int dir)
+static struct sof_ipc_dai_config *hda_dai_update_config(struct snd_soc_dapm_widget *w,
```

Comment:
IIUC `w` is not just some widget, it's a DAI widget? Wouldn't it be better to call it here and in all other functions, where you add it, `dai_widget`? Maybe this is obvious though in which case feel free to ignore.

Reply:
its the playback/capture widget associated with a CPU DAI. So I think its quite obvious that it is a DAI widget. So I'll leave it as is for now